nautobot/core/testing/views.py

```diff
@@ -442,6 +443,10 @@ def test_edit_object_with_permission(self):
             # Try GET with model-level permission
             self.assertHttpStatus(self.client.get(self._get_url("edit", instance)), 200)
 
+            # We cannot edit IPAddress's host field once it is already set.
+            # but we do need the address field to be not empty
+            if isinstance(instance, IPAddress):
+                self.form_data["address"] = instance.address
```

Comment:
I'm not a big fan of having single-model-specific logic in the core generic tests. Can we solve this in a more flexible way? IIRC @bryanculver had some similar challenges in #3837 (see https://github.com/nautobot/nautobot/pull/3837/files#diff-63390a9316fbabcd9c87d7ea0980d9ff2738d5da7d7def31e8f152d5fcf92996R627) that were worked around by overloading these two test methods locally. That'd be my preferred approach for this PR; longer term I think we should look at borrowing the pattern that already exists in `nautobot.core.testing.api` where by default (as today) the same data is used for both create and edit tests, but you have the option to override one or the other if needed.